cmd/exec/exec.go

```diff
@@ -77,22 +53,136 @@ func getCommandFromName(config config.OneBuildConfiguration, cmd string) string
 	return ""
 }
 
-func createMapOfCommandsToExecute(oneBuildConfiguration config.OneBuildConfiguration, commands ...string) (map[string]string, error) {
-	var cmdMap map[string]string
-	cmdMap = make(map[string]string)
+func bashCommand(s *sh.Session, command string) *sh.Session {
+	return s.Command("bash", "-c", command)
+}
+
+func executeAndStopIfFailed(command *CommandContext) {
+	err := command.commandSession.Run()
+	if err != nil {
+		exitCode := (err.Error())[12:]
+		utils.PrintlnDashedErr("Execution failed during phase \"" + command.name + "\" - Execution of the script \"" + command.command + "\" returned non-zero exit code : " + exitCode)
+		utils.ExitWithCode(exitCode)
+	}
+}
+
+func buildExecutionPlan(onebuildConfig config.OneBuildConfiguration, commands ...string) OneBuildExecutionPlan {
+
+	before := onebuildConfig.Before
+	var executionPlan OneBuildExecutionPlan
+	if before != "" {
+		executionPlan.Init = &CommandContext{"before", before, bashCommand(sh.NewSession(), before)}
+	}
 
 	for _, name := range commands {
-		value := getCommandFromName(oneBuildConfiguration, name)
-		if value == "" {
-			utils.Println("No command '" + name + "' found in config file\n")
-			config.PrintConfiguration(oneBuildConfiguration)
-			return nil, errors.New("")
+		executionCommand := getCommandFromName(onebuildConfig, name)
+		if executionCommand == "" {
+			utils.PrintlnErr("Error building exectuion plan. Command \"" + name + "\" not found.")
+			config.PrintConfiguration(onebuildConfig)
+			utils.Exit(1)
```

Comment:
To follow Unix status code - Let's return `exit code 127` for command not found.